src/code/z_kankyo.c

```diff
@@ -667,27 +980,289 @@ Color_RGBA8 sSkyboxEnvColors[] = {
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if ((((void)0, gSaveContext.skyboxTime) >= CLOCK_TIME(6, 0)) ||
+        (((void)0, gSaveContext.save.time) < CLOCK_TIME(6, 0)) ||
+        (((void)0, gSaveContext.save.time) >= (CLOCK_TIME(6, 0) + 0x10))) {
+        gSaveContext.skyboxTime = gSaveContext.save.time;
+    }
+
+    time = ((void)0, gSaveContext.save.time);
+
+    if ((time >= CLOCK_TIME(18, 0)) || (time < CLOCK_TIME(6, 0))) {
+        gSaveContext.save.isNight = true;
+    } else {
+        gSaveContext.save.isNight = false;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_UpdateNextDayTime.s")
+void func_800F6CEC(PlayState* play, u8 arg1, AdjLightSettings* adjLightSettings, EnvLightSettings* lightSettings) {
+    s32 phi_t1;
+    s32 temp_v1_2;
+    s32 temp_v1 = (arg1 % 4);
+
+    if ((((void)0, gSaveContext.save.day) >= 2) && (arg1 >= 4) && (arg1 < 8)) {
+        temp_v1_2 = (((void)0, gSaveContext.save.day) * 4) + 4;
+        for (phi_t1 = 0; phi_t1 != 3; phi_t1++) {
+            adjLightSettings->ambientColor[phi_t1] =
+                lightSettings[temp_v1_2 + temp_v1].ambientColor[phi_t1] - lightSettings[temp_v1].ambientColor[phi_t1];
+            adjLightSettings->light1Color[phi_t1] =
+                lightSettings[temp_v1_2 + temp_v1].light1Color[phi_t1] - lightSettings[temp_v1].light1Color[phi_t1];
+            adjLightSettings->light2Color[phi_t1] =
+                lightSettings[temp_v1_2 + temp_v1].light2Color[phi_t1] - lightSettings[temp_v1].light2Color[phi_t1];
+            adjLightSettings->fogColor[phi_t1] =
+                lightSettings[temp_v1_2 + temp_v1].fogColor[phi_t1] - lightSettings[temp_v1].fogColor[phi_t1];
+        }
+        adjLightSettings->fogNear =
+            lightSettings[temp_v1_2 + temp_v1].blendRateAndFogNear - lightSettings[temp_v1].blendRateAndFogNear;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_UpdateTime.s")
+    if ((arg1 >= 4) && (arg1 < 8) && (gWeatherMode == WEATHER_MODE_1)) {
+        adjLightSettings->ambientColor[0] = -50;
+        adjLightSettings->ambientColor[1] = -100;
+        adjLightSettings->ambientColor[2] = -100;
+        adjLightSettings->light1Color[0] = -100;
+        adjLightSettings->light1Color[1] = -100;
+        adjLightSettings->light1Color[2] = -100;
+        adjLightSettings->light2Color[0] = -100;
+        adjLightSettings->light2Color[1] = -100;
+        adjLightSettings->light2Color[2] = -100;
+
+        temp_v1 = arg1;
+        adjLightSettings->fogColor[0] = -lightSettings[temp_v1].fogColor[0] + 30;
+        adjLightSettings->fogColor[1] = -lightSettings[temp_v1].fogColor[1] + 30;
+        adjLightSettings->fogColor[2] = -lightSettings[temp_v1].fogColor[2] + 45;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_800F6CEC.s")
+u8 func_800F6EA4(f32 arg0, f32 arg1, f32 arg2) {
```

Comment:
Wonder if we should just name this `Environment_LerpColor` or something?

Reply:
There's already a `Environment_LerpRGB8`, but since that's specifically for the `Color_RGB8`, maybe it's fine to have this function be `Environment_LerpColor`?